### PR TITLE
Add shift click support for plannings

### PR DESCRIPTION
**Pull request by fmeynard**
Close https://github.com/crf-devs/resop/issues/101

### CODE REVIEW

File: assets/js/planning-update.js [hunk 1]
```diff
@@ -122,8 +122,85 @@ function checkLastUpdate(forceUpdate) {
...
+
+  $(document).on('keydown', function (e) {
+    if (e.shiftKey && $lastClickedTd && !$lastClickedTd.hasClass('highlight')) {
+      $planning.find('.highlight').removeClass('highlight');
+      $lastClickedTd.addClass('highlight');
+    }
+  });
+
+  $(document).on('keyup', function () {
+    $planning.find('.highlight').removeClass('highlight');
```
Comment (mRoca) [inline assets/js/planning-update.js:202]: you should check the used key here, as `$planning.find('.highlight')` can take some time

File: assets/js/planning-update.js [hunk 2]
```diff
@@ -122,8 +122,85 @@ function checkLastUpdate(forceUpdate) {
...
+      .find('td')
+      .filter((i, td) => Date.parse($(td).data('from')) >= tdFrom && Date.parse($(td).data('to')) <= tdTo)
+      .each(function () {
+        if (isChecked !== $(this).hasClass('checked')) {
+          selectTableBox($(this));
+        }
+      });
+  };
+
+  $planning
```
Comment (mRoca) [inline assets/js/planning-update.js:184]: this line should be inside `$(document).ready(function () {`

File: assets/js/planning-update.js (hunk 1 above)
Comment (mRoca) [reply assets/js/planning-update.js:202]: And this `on('keyup'` doesn't work if I don't release the shift key

File: assets/css/availability-form.scss [hunk 3]
```diff
@@ -28,6 +28,10 @@ table.availability-table {
       }
     }
 
+    &.highlight {
+      box-shadow: inset 0 0 6px 1px black;
```
Comment (mRoca) [inline assets/css/availability-form.scss:32]: `inset 0 0 0 4px crimson`

File: assets/js/planning-update.js [hunk 4]
```diff
@@ -122,23 +148,82 @@ function checkLastUpdate(forceUpdate) {
...
 
+  $planning.on('click', '.slot-box', handleSlotBoxClick);
+  $(document).on('keydown keyup', handleShiftInput);
+
   let urlParams = new URLSearchParams(window.location.search);
   if (urlParams.has('scrollTop')) {
     $(window).scrollTop(urlParams.get('scrollTop'));
   }
 
-  $planning.on('click', '.slot-box input:checkbox', function (e) {
```
Comment (mRoca) [inline assets/js/planning-update.js:133]: can you revert this line ? Without it we cannot unselect a box by clicking on the checkbox

File: assets/js/planning-update.js [hunk 5]
```diff
@@ -20,6 +20,32 @@ function selectTableBox($tableBox) {
...
+  const $slotBox = event.target === 'input' ? $(this).closest('.slot-box') : $(this);
+  selectTableBox($slotBox);
+
+  const $planning = $('.planning');
+  const $lastClickedTd = $planning.find('.prev-clicked');
+
+  if ($lastClickedTd && event.shiftKey) {
+    handleShiftClick($planning, $slotBox, $lastClickedTd);
+    $lastClickedTd.removeClass('prev-clicked');
+    $lastClickedTd.removeClass('highlight');
```
Comment (mRoca) [inline assets/js/planning-update.js:35]: I would have added this class on the table, or on the body, as `.shift-pressed`

File: assets/js/planning-update.js [hunk 6]
```diff
@@ -20,6 +20,32 @@ function selectTableBox($tableBox) {
...
+  }
+
+  $slotBox.addClass('prev-clicked');
```
Comment (mRoca) [inline assets/js/planning-update.js:38]: you must remove old `prev-clicked` even if the shift key is not currently pressed

File: assets/js/planning-update.js (hunk 5 above)
Comment (fmeynard) [reply assets/js/planning-update.js]: Done at table level

File: assets/css/availability-form.scss [hunk 7]
```diff
@@ -28,21 +26,9 @@ table.availability-table {
...
-    &.disabled {
-      cursor: default;
-    }
-
     &.booked {
       background-color: #67a2d3 !important;
     }
-
-    &.locked {
-      background-color: #7a7a7a !important;
```
Comment (mRoca) [inline assets/css/availability-form.scss:40]: `!important` must be kept

**Pull request merged**